.azure-pipelines/sign-binary.yaml

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger: none
+
+pool:
+  name: Azure Pipelines
+  vmImage: windows-latest
+
+variables:
+  - name: ProductBinPath
+    value: '$(System.DefaultWorkingDirectory)\assets'
+  - name: sourceRepository
+    value: 'microsoftgraph/msgraph-cli'
+  - name: repositoryConnection
+    value: 'GitHub - calebkiage'
+  - name: zipTemplate
+    value: msgraph-cli-win-x64-{0}.zip
+parameters:
+  - name: tag
+    displayName: Tag
+    type: string
+    default: latest
+
+
+# download -> scan -> zip -> upload
+#          -> sign
+#
+
+stages:
+- stage: download
+  jobs:
+    - job: download
+      steps:
+      - checkout: none
+      # Download GitHub Release
+      # Downloads a GitHub Release from a repository
+      - task: DownloadGitHubRelease@0
+        inputs:
+          connection: $(repositoryConnection)
+          userRepository: $(sourceRepository)
+          defaultVersionType: 'specificTag' # Options: latest, specificVersion, specificTag
+          version: ${{ parameters.tag }} # Required when defaultVersionType != Latest
+          itemPattern: '*-win-x64-*' # Optional
+          downloadPath: '$(ProductBinPath)'
+      - publish: $(ProductBinPath)
+        artifact: download-zip
+    - job: version
+      dependsOn: []
+      pool:
+        vmImage: ubuntu-latest
+      steps:
+      - checkout: none
+      - ${{ if startsWith(parameters.tag, 'v') }}:
+        - script: echo "##vso[task.setvariable variable=cliVersion;isOutput=true]$(echo ${{ parameters.tag }} | cut -c 2-)"
+          name: setCliVersionStep
+      - ${{ else }}:
+        - script: echo "##vso[task.setvariable variable=cliVersion;isOutput=true]${{ parameters.tag }}"
+          name: setCliVersionStep
+    - job: unzip
+      dependsOn: [download, version]
+      variables:
+        cliVersion: $[ dependencies.version.outputs['setCliVersionStep.cliVersion'] ]
+        cliArchiveNameVar: $[format(variables.zipTemplate, variables.cliVersion)]
+      pool:
+        vmImage: ubuntu-latest
+      steps:
+      - checkout: none
+      - download: current
+        artifact: download-zip
+      - script: echo "##vso[task.setvariable variable=cliArchiveName;isOutput=true]$(cliArchiveNameVar)"
+        name: setArchiveNameStep
+      - script: unzip -j "$(Pipeline.Workspace)/download-zip/$(setArchiveNameStep.cliArchiveName)" -d $(Pipeline.Workspace)/download-zip/unzipped
+      - publish: $(Pipeline.Workspace)/download-zip/unzipped
+        artifact: unzipped
```

Comment:
booting up agents can take a while, and you might get pushed to the end of the queue between two jobs/stages. I'd argue download/version/unzip should be a single job.

Reply:
This is a good point. I initially had trouble with setting the archive name variable, but I'm willing to try again